Add support for audio and video tags in epub3 and add firstImageIsCover option

uriesk opened the pull request:
Epub3 supports `<video>` and `<audio>` tags. This pull request adds support for them.
The media gets downloaded the same as images do.

Add `firstImageIsCover` option. When true, the first appearing image inside the ebook will get marked as cover in the manifest, so epub viewers can use it as thumbnail.

Caveats

- Audio and Video files get put into an `audiovideo` subfolder, which got [recommended here](https://ebookflightdeck.com/handbook/media).
- I change occuring `controls` attributes to `controls="Controls"`, cause this is what XHTML wants. `controls="controls"` didn't work, cause rehype-stringify would change it back, but XHTML isn't case sensitive in attributes, so it works.
- an additional little commit, to not add `<meta name="cover" content="image_cover"/` in the manifest if there is no cover (calibre ebook editor gave a warning for this).
- add start of content to guide
- additional  ad `nav landmarks` to epub3, since `<guide>` is deprecated and [at least Apple has the requirement for landmarks to exist](https://www.heliconbooks.com/?id=blog&postid=EPUB3NavigationDocument).

uriesk commented:
@e-adrien hi, i am now ready with this PR. I decided to add multiple commits for different issues to it, because it would be too cumbersome otherwise, as everything happens in only three files.

I did check with epubcheck-5.2.0 and with the check of calibre ebook editor, to make sure that everything is valid.

e-adrien commented:
Hi @uriesk, thank you for this PR.

I think it will be better to use `useFirstImageAsCover` for the parameter name instead of `firstImageIsCover`.
I also think that the Audio and Video files embedding should be an optional feature (`downloadAudioVideoFiles` parameter, `false` by default).

Do you mind adding a test [here](https://github.com/lesjoursfr/html-to-epub/blob/6e68bcde98b269659b14fab96440ace99223ad0e/tests/epub.spec.ts) ? 

Thank you again for your work.

e-adrien commented on templates/epub2/content.opf.ejs:
```diff
@@ -13,7 +13,7 @@
         <dc:date opf:event="modification"><%= date %></dc:date>
         <dc:language><%= lang || "en" %></dc:language>
-        <meta name="cover" content="image_cover" />
+        <% if(locals.cover || (locals.firstImageIsCover && images.length)) { %><meta name="cover" content="<%= locals.cover ? 'image_cover' : 'image_0' %>"/><% } %>
```
I wonder if we can avoid these `if` by puting some logic in this method https://github.com/lesjoursfr/html-to-epub/blob/6e68bcde98b269659b14fab96440ace99223ad0e/src/index.ts#L693

uriesk commented:
@e-adrien added a bunch of test cases, renamed to `useFirstImageAsCover` and added `downloadAudioVideoFiles`

I don't like the long condition in the template either, but i didn't want to touch the `makeCover` stuff because it does so much extra things with creating a separate page for the cover.
Feel free to refactor it, but i personally think it's good how it is now.

e-adrien commented:
Thank you. I'll try to have this merged by the end of this month.
Thank you again for your work.
